OpenRA.Mods.Common/Traits/Mobile.cs

```diff
@@ -110,7 +113,20 @@ public bool CanEnterCell(World world, Actor self, CPos cell, SubCell subCell = S
 			if (locomotor.MovementCostForCell(cell) == short.MaxValue)
 				return false;
 
-			return locomotor.CanMoveFreelyInto(self, cell, subCell, check, ignoreActor);
+			if (!locomotor.CanMoveFreelyInto(self, cell, subCell, check, ignoreActor))
+				return false;
+
+			// We cannot cross diagonals if another actor is already moving diagonal.
+			if (!CheckAdjacentOnDiagonal || !fromCell.HasValue)
```

Comment:
locomotor.Info.CheckAdjacentOnDiagonal, and then the `CheckAdjacentOnDiagonal` defined here on `MobileInfo` can be removed.